lib/ueberauth/strategy/oidc.ex

```diff
@@ -98,15 +111,23 @@ defmodule Ueberauth.Strategy.OIDC do
     conn
     |> put_private(:ueberauth_oidc_opts, nil)
     |> put_private(:ueberauth_oidc_tokens, nil)
+    |> put_private(:ueberauth_oidc_user_info, nil)
   end
 
   @doc """
   Returns the configured uid field from the id token.
   """
   def uid(conn) do
     private = conn.private
-    uid_field = option(private.ueberauth_oidc_opts, :uid_field)
-    scrub_value(private.ueberauth_oidc_tokens[:id][:claims][uid_field])
+
+    with user_info <- option(private.ueberauth_oidc_opts, :user_info),
```

Comment:
maybe change `:user_info` into something that's closer to what it does, it makes sure that another fetch has to be done